numpy/core/src/private/npy_longdouble.c

```diff
@@ -0,0 +1,99 @@
+#include <Python.h>
+
+#define NPY_NO_DEPRECATED_API NPY_API_VERSION
+#include "numpy/ndarraytypes.h"
+#include "numpy/npy_math.h"
+
+/* This is a backport of Py_SETREF */
+#define NPY_SETREF(op, op2)                      \
+    do {                                        \
+        PyObject *_py_tmp = (PyObject *)(op);   \
+        (op) = (op2);                           \
+        Py_DECREF(_py_tmp);                     \
+    } while (0)
+
+
+/* Heavily derived from PyLong_FromDouble
+ * Notably, we can't set the digits directly, so have to shift and or instead.
+ */
+PyObject *
+npy_longdouble_to_PyLong(npy_longdouble ldval)
+{
+    PyObject *v;
+    PyObject *l_chunk_size;
+    // number of bits to extract at a time. CPython uses 30, but that's because
```

Comment:
Hmm, should be C style comment.

Reply:
Careless, good spot. This is the reason for the Mac failure?

Reply:
No. At the moment I'm suspecting the power function.

Reply:
The problem is this line: `huge_ld = 1234 * np.longdouble(2) ** exp`. We need someone running on a Mac to check this out. Maybe a fallback double function is being used somewhere.

Reply:
Here it is, `powl` is blacklisted on the Mac
```
/* powl gives zero division warning on OS X, see gh-8307 */
#if defined(HAVE_POWL) && defined(NPY_OS_DARWIN)
#undef HAVE_POWL
#endif
```

Reply:
Although I don't see why that should be a problem, the double value of the fallback function should be OK with result cast to `long double`, so the multiplication should work. Maybe inlining is failing somewhere ...

Reply:
Fired up the old Mac. The Mac double `pow` function errors with `exp == 1024`, Linux works with that. I'm simply going to use `exp = 1023` as a fix in this case.

Reply:
exp=1023 doesn't work, because then the result fits in `double` - the goal here is to produce a longdouble that exceeds the limits for a double.

Reply:
Yes, that is what I did. See #10000 . Oh, looky there, issue 10000, it's like watching the odometer turn over ...

Reply:
Guess we will eventually find out if this all works for IBM double double.